fix(auth): automatically switch auth type if an API token is provided

PeterSchafer opened the pull request.

thisislawatts commented on pkg/local_workflows/auth_workflow.go:
```diff
@@ -57,19 +56,33 @@ func OpenBrowser(authUrl string) {
 	// get necessary objects from invocation context
 	config := invocationCtx.GetConfiguration()
-	logger := invocationCtx.GetLogger()
+	logger := invocationCtx.GetEnhancedLogger()
```
**praise:** Nice upgrade

thisislawatts reviewed (approved):
Good to get some tests introduced around the auth_workflow

thisislawatts commented on pkg/configuration/constants.go:
```diff
@@ -22,6 +22,7 @@ const (
 	RAW_CMD_ARGS                    string = "internal_raw_cmd_args"
 	WEB_APP_URL                     string = "internal_snyk_app"
 	INPUT_DIRECTORY                 string = "targetDirectory"
+	UNNAMED_PARAMETER               string = INPUT_DIRECTORY
```
**nitpick(non-blocking):** It may be useful to move the comment around positional arguments to this declaration to improve readability.

cmars replied:
I'd go further and fix this... it's more than a nit to me.

cmars commented on pkg/configuration/constants.go:
```diff
@@ -22,6 +22,7 @@ const (
 	RAW_CMD_ARGS                    string = "internal_raw_cmd_args"
 	WEB_APP_URL                     string = "internal_snyk_app"
 	INPUT_DIRECTORY                 string = "targetDirectory"
+	UNNAMED_PARAMETER               string = INPUT_DIRECTORY                       //UNNAMED_PARAMETER in the CLI is the positional argument
```
It seems weird that the value for this is `"targetDirectory"` when it's not a target directory.

Feels the wrong way around. The value should be `positionalArgument` and the context of that might be a target directory if that's what `positionalArgument` means to a specific command.

cmars replied:
It might be arbitrary now in the way it works, but can easily create bugs or hard to debug situations when we have to reverse the meaning.

PeterSchafer replied:
Naming is hard and especially in this case. The problem with positional arguments is, that it is something from the cli domain as it describes the concept of arguments that have a meaning based on their position. In a framework that doesn't necessarily know about the application it is used in, the best naming would be usage specific. A mapping in the cli application of the usage based name to the positional argument would be the best. I will adapt the PR accordingly.
Thank you!

cmars reviewed (commented):
Concerned with the conflation of unnamed parameter & input directory in the constants.

PeterSchafer commented:
I addressed the previously expressed concerns about the variable naming and refactored the logic a bit to be more clear and support better debug logging.
There is the associated CLI PR here https://github.com/snyk/cli/pull/5240

j-luong commented on pkg/local_workflows/auth_workflow.go:
```diff
@@ -57,37 +58,42 @@ func OpenBrowser(authUrl string) {
+			return err
 		}
 
 		fmt.Println(auth.AUTHENTICATED_MESSAGE)
```
question: can we not use the logger here?
